bot_positions:
  ros__parameters:

    controller_name: "scaled_joint_trajectory_controller"
    wait_sec_between_publish: 5

    goal_names: [
      "home",
      # "chip_home_1",
      "chip_home_2",
      # "chip_home_3",
      # "chip_home_4",
      # "above_tray_1",
      # "above_tray_2",
      # "chip_place_1",
      # "chip_place_2",
      # "bottom_case_place_1",
      # "bottom_case_place_2",
      # "top_case_place_1",
      # "top_case_place_2",
      # "battery_place_1",
      # "battery_place_2",
      "chip_1",
      "chip_2",
      "chip_3",
      "chip_4",
      "chip_5",
      "chip_6",
      "chip_7",
      "chip_8",
      "chip_9",
      "chip_10",
      "chip_11",
      "chip_12",
      "chip_13",
      "chip_14",
      "chip_15",
      "chip_16",
      "chip_17",
      "chip_18",
      "chip_19",
      "chip_20",
      "chip_21",
      "chip_22",
      "chip_23",
      "chip_24",
      "chip_25",
      "chip_26",
      "chip_27",
      "chip_28",
      "chip_29",
      "chip_30",
      "chip_31",
      "chip_32",
      "chip_33",
      "chip_34",
      "chip_35",
      "chip_36",
      "chip_37",
      "chip_38",
      "chip_39",
      "chip_40",
      "chip_41",
      "chip_42",
      "chip_43",
      "chip_44",
      "chip_45",
      "chip_46",
      "chip_47",
      "chip_48",
      # "case_pick_home",
      # "case_1",
      # "case_2",
      # "case_3",
      # "case_4",
      # "case_5",
      # "case_6",
      # "case_7",
      # "case_8",
      # "case_9",
      # "case_10",
      # "case_11",
      # "case_12",
      # "case_13",
      # "case_14",
      # "case_15",
      # "case_16",
      # "case_17",
      # "battery_pick",
      # "battery_pick_home",
      # "tray_pick_1",
      # "tray_pick_1_above",
      # "tray_pick_2",
      # "tray_pick_2_above",
      # "tray_unload",
      # "tray_unload_above",
    ]
    home:
      positions: [-1.568876465,-2.34362812,2.104518012,-1.361007751,-1.565385806,-4.708374723]
    # chip_home_1:
    #   positions: [-0.814021563,-1.142841594,0.80965824,-1.513898593,-1.734857276,-5.58644987]
    chip_home_2:
      positions: [-0.841597765,-1.259604121,0.827286065,-1.433438915,-1.824916266,-5.554510344]
    # chip_home_3:
    #   positions: [-0.986983692,-1.41703282,0.952426173,-1.436056909,-1.741838593,-5.714207971]
    # chip_home_4:
    #   positions: [-1.133765882,-1.519483647,1.050339144,-1.403070186,-1.686337123,-5.886646501]
    # above_tray_1:
    #   positions: [-0.165806279,-1.905375944,2.046922147,-1.721243708,-1.589820416,-3.32275783]
    # above_tray_2:
    #   positions: [-0.855734932,-1.791580477,1.983392162,-1.799434459,-1.596976266,-3.99697852]
    # chip_place_1:
    #   positions: [-0.074525559,-1.862789911,2.56144521,-2.259328717,-1.55788089,-3.239505625]
    # chip_place_2:
    #   positions: [-0.967436004,-1.800307123,2.517462913,-2.322335103,-1.544616388,-4.122816759]
    # bottom_case_place_1:
    #   positions: [-0.290771853,-1.465552973,2.326000294,-2.44031936,-1.595929068,-3.412467753]
    # bottom_case_place_2:
    #   positions: [-0.826587934,-1.291543646,2.111150263,-2.412743158,-1.587376955,-3.948458367]
    # top_case_place_1:
    #   positions: [-0.012915436,-1.419476281,2.263168441,-2.415361152,-1.597674397,-3.134611337]
    # top_case_place_2:
    #   positions: [-0.644550093,-1.389107552,2.255488992,-2.455678258,-1.590867613,-3.766420526]
    # battery_place_1:
    #   positions: [-0.471762497,-1.822123739,2.641904889,-2.363873939,-1.581617368,-2.03697377]
    # battery_place_2:
    #   positions: [-1.088212789,-1.51546939,2.375044046,-2.417281014,-1.594707338,-2.65272593]
    chip_1:
      positions: [-0.832172987,-0.888721655,0.618021088,-1.55788089,-1.851270738,-5.568298446]
    chip_2:
      positions: [-0.822748209,-0.946666586,0.728849496,-1.596452667,-1.819680278,-5.570043775]
    chip_3:
      positions: [-0.792903079,-1.039692635,0.907047612,-1.709724535,-1.8245672,-5.539151447]
    chip_4:
      positions: [-0.770562865,-1.057320461,0.931133156,-1.664171442,-1.814095225,-5.513320574]
    chip_5:
      positions: [-0.748746249,-1.076170017,0.959756556,-1.631010186,-1.796292866,-5.487489701]
    chip_6:
      positions: [-0.715759526,-1.142317995,1.091703447,-1.724559834,-1.802750584,-5.45397938]
    chip_7:
      positions: [-0.931133156,-1.006356847,0.789761487,-1.612684229,-1.689653249,-5.699721738]
    chip_8:
      positions: [-0.897099236,-1.018225086,0.807214779,-1.562069681,-1.717927583,-5.644394801]
    chip_9:
      positions: [-0.879995009,-1.021017612,0.807040246,-1.493129175,-1.692969374,-5.602681432]
    chip_10:
      positions: [-0.853989603,-1.09606677,0.946492053,-1.56922553,-1.699776159,-5.616469533]
    chip_11:
      positions: [-0.834616448,-1.139700002,1.0262536,-1.600466924,-1.682846465,-5.578944954]
    chip_12:
      positions: [-0.809832773,-1.177573646,1.096241303,-1.596801733,-1.682497399,-5.526759609]
    chip_13:
      positions: [-1.027824396,-1.048419282,0.82100288,-1.534668011,-1.568701932,-5.803394296]
    chip_14:
      positions: [-0.974068256,-1.048070216,0.821177413,-1.486322391,-1.648463479,-5.734279257]
    chip_15:
      positions: [-0.949808179,-1.060287521,0.844739358,-1.437627705,-1.658411855,-5.684362841]
    chip_16:
      positions: [-0.917345055,-1.17722458,1.056796862,-1.589820416,-1.651954137,-5.682792044]
    chip_17:
      positions: [-0.880344075,-1.230631656,1.164483677,-1.65212867,-1.676388747,-5.634446424]
    chip_18:
      positions: [-0.848579082,-1.295732437,1.275137552,-1.714436924,-1.69331844,-5.586100804]
    chip_19:
      positions: [-1.035678378,-1.092576112,0.889245254,-1.54356919,-1.651779604,-5.768836777]
    chip_20:
      positions: [-1.003389787,-1.140398133,0.971799328,-1.555437429,-1.669232897,-5.745449365]
    chip_21:
      positions: [-0.997106602,-1.182111502,1.039518102,-1.548281579,-1.645147353,-5.745449365]
    chip_22:
      positions: [-0.97197386,-1.254717199,1.167799803,-1.624203402,-1.644798287,-5.745449365]
    chip_23:
      positions: [-0.94631752,-1.309171472,1.262920247,-1.668185699,-1.644449221,-5.685933637]
    chip_24:
      positions: [-0.910014672,-1.347568715,1.331686219,-1.679704872,-1.672374489,-5.637937083]
    chip_25:
      positions: [-1.067268838,-1.271646893,1.213352896,-1.873785485,-1.699950691,-5.806710421]
    chip_26:
      positions: [-1.040216234,-1.308647873,1.264141977,-1.824741733,-1.70047429,-5.807408553]
    chip_27:
      positions: [-1.049117413,-1.394518072,1.405862712,-1.907819405,-1.648463479,-5.800776302]
    chip_28:
      positions: [-1.023635606,-1.410749634,1.420348945,-1.852317935,-1.64793988,-5.800776302]
    chip_29:
      positions: [-0.976337184,-1.484402529,1.552993968,-1.956165026,-1.696983632,-5.721189288]
    chip_30:
      positions: [-0.941081533,-1.487893187,1.549328777,-1.880243203,-1.713913325,-5.704957726]
    chip_31:
      positions: [-1.130973355,-1.282467934,1.202182789,-1.748470845,-1.665567705,-5.889962627]
    chip_32:
      positions: [-1.090132651,-1.358215224,1.340412866,-1.845511151,-1.714087858,-5.849296455]
    chip_33:
      positions: [-1.096939435,-1.433089849,1.445307154,-1.880243203,-1.655619328,-5.810375613]
    chip_34:
      positions: [-1.062381916,-1.395216204,1.363451212,-1.709026404,-1.683195531,-5.782624878]
    chip_35:
      positions: [-1.066919772,-1.439023968,1.423316005,-1.692969374,-1.62228354,-5.794842182]
    chip_36:
      positions: [-1.026777199,-1.546361717,1.607273708,-1.867327767,-1.657015592,-5.779308752]
    chip_37:
      positions: [-1.196772268,-1.340412866,1.261000385,-1.739395133,-1.643576557,-5.947558492]
    chip_38:
      positions: [-1.17896991,-1.394169006,1.349663111,-1.764527874,-1.640958563,-5.928534403]
    chip_39:
      positions: [-1.156629695,-1.459095255,1.451590339,-1.82037841,-1.645496419,-5.904274327]
    chip_40:
      positions: [-1.12765723,-1.516342054,1.542521993,-1.857902989,-1.667836633,-5.86273549]
    chip_41:
      positions: [-1.107760476,-1.551772238,1.58353723,-1.830326787,-1.662426113,-5.83882448]
    chip_42:
      positions: [-1.08053334,-1.631010186,1.706932008,-1.916546052,-1.674643417,-5.822418385]
    chip_43:
      positions: [-1.267109037,-1.409353371,1.360833218,-1.793500339,-1.612858762,-5.999045705]
    chip_44:
      positions: [-1.252448271,-1.433962513,1.389980216,-1.755277629,-1.612684229,-5.983512275]
    chip_45:
      positions: [-1.217018087,-1.519134581,1.52384697,-1.846383816,-1.653524933,-5.958903132]
    chip_46:
      positions: [-1.188220155,-1.586678823,1.61809475,-1.886700921,-1.67359622,-5.932723193]
    chip_47:
      positions: [-1.193107077,-1.630835653,1.673770753,-1.875356281,-1.636769773,-5.926614541]
    chip_48:
      positions: [-1.145983187,-1.688955117,1.748645378,-1.89734743,-1.67481795,-5.886821034]
    # case_pick_home:
    #   positions: [-1.153662636,-2.14797671,2.034879375,-2.452536665,-1.949009176,-6.04948572]
    # case_1:
    #   positions: [-1.15872409,-2.112546527,2.371204322,-2.824815394,-1.945343984,-6.05123105]
    # case_2:
    #   positions: [-1.184554963,-2.07275302,2.340486527,-2.83947616,-1.923876434,-6.066066348]
    # case_3:
    #   positions: [-1.218239818,-2.038893632,2.295457032,-2.834763771,-1.895427568,-6.085439503]
    # case_4:
    #   positions: [-1.238311104,-2.012713693,2.288824781,-2.855358656,-1.890366113,-6.088755629]
    # case_5:
    #   positions: [-1.251575607,-1.984613892,2.254092729,-2.853438794,-1.867327767,-6.103939993]
    # case_6:
    #   positions: [-1.260825852,-1.932079482,2.224073066,-2.877524338,-1.859648318,-6.109001448]
    # case_7:
    #   positions: [-1.293114443,-1.9099138,2.191435409,-2.871939284,-1.832246649,-6.126629273]
    # case_8:
    #   positions: [-1.303586418,-1.875530814,2.159670416,-2.875953541,-1.82334547,-6.13238886]
    # case_9:
    #   positions: [-1.320341579,-1.842544091,2.122669436,-2.874208212,-1.809208303,-6.141290039]
    # case_10:
    #   positions: [-1.334478746,-1.805892177,2.09666403,-2.886774583,-1.797165531,-6.148794955]
    # case_11:
    #   positions: [-1.344601656,-1.770636526,2.070135026,-2.89672296,-1.788613417,-6.154205475]
    # case_12:
    #   positions: [-1.359960553,-1.7392206,2.045525883,-2.905449606,-1.775523448,-6.162408523]
    # case_13:
    #   positions: [-1.360135086,-1.705710278,1.986533755,-2.879793266,-1.775523448,-6.162583056]
    # case_14:
    #   positions: [-1.368338134,-1.676737812,1.961750079,-2.884854721,-1.768542131,-6.166771846]
    # case_15:
    #   positions: [-1.375144918,-1.638864168,1.910437399,-2.872113817,-1.762782545,-6.170437038]
    # case_16:
    #   positions: [-1.384918761,-1.604481181,1.877974275,-2.87525541,-1.754579497,-6.175673025]
    # case_17:
    #   positions: [-1.391900078,-1.576032315,1.829454122,-2.855882255,-1.748470845,-6.179338217]
    # battery_pick:
    #   positions: [-2.452711198,-2.630211183,2.383072561,-1.838529834,-1.012290966,-3.882484921]
    # battery_pick_home:
    #   positions: [-2.452536665,-2.589545011,2.237163035,-1.645147353,-1.067617903,-3.893305962]
    # tray_pick_1:
    #   positions: [-0.156730567,-1.690176848,2.435956037,-2.343977185,-1.624203402,-3.29832322]
    # tray_pick_1_above:
    #   positions: [-0.156730567,-1.789486082,2.36212861,-2.170840524,-1.624028869,-3.299021352]
    # tray_pick_2:
    #   positions: [-0.893608577,-1.564513141,2.338217599,-2.379058303,-1.54112573,-4.028568979]
    # tray_pick_2_above:
    #   positions: [-0.893608577,-1.741314995,2.179218104,-2.043082422,-1.540776664,-4.02979071]
    # tray_unload:
    #   positions: [-0.309446876,-1.000073661,1.541474795,-2.123367568,-1.575683249,-3.441614752]
    # tray_unload_above:
    #   positions: [-0.309446876,-1.086641992,1.440769298,-1.935919206,-1.575683249,-3.441963818]

    joints:
      - shoulder_pan_joint
      - shoulder_lift_joint
      - elbow_joint
      - wrist_1_joint
      - wrist_2_joint
      - wrist_3_joint

    check_starting_point: true
    starting_point_limits:
      shoulder_pan_joint: [-2.0,2.0]
      shoulder_lift_joint: [-3.0,-1.0]
      elbow_joint: [-0.1,3.0]
      wrist_1_joint: [-3.0,0.0]
      wrist_2_joint: [-3.0,0.1]
      wrist_3_joint: [-7.0,0.1]
